sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/transform/BeamSetOperatorsTransforms.java

```diff
@@ -97,8 +97,10 @@ public void processElement(ProcessContext ctx) {
           }
           break;
         case MINUS:
-          // Say for Row R, there are m instances on left and n instances on right,
-          // EXCEPT ALL outputs MAX(m - n, 0) instances of R.
+          // Say for Row R, there are m instances on left and n instances on right:
+          // - EXCEPT ALL outputs MAX(m - n, 0) instances of R.
+          // - EXCEPT DISTINCT and EXCEPT output a single instance of R if m > 0 and n == 0, else
```

Comment:
According to Calcite SQL doc( https://calcite.apache.org/docs/reference.html), 

ALL is the default for UNION/INTERSECT/DISTINCT. So you can write something like
```
EXCEPT [ALL] 
EXCEPT DISTINCT
```

Reply:
Ah good point. That seems to be at odds with the test case I modified though - the SQL query in `testExcept` just says `EXCEPT` and seems to run with `all=false`. Any idea what's going on there?

Reply:
That's interesting. I am thinking if Calcite doc is stale. 

Could you check the logical plan and see what's default setting is?

Reply:
Discussed this a bit offline. It looks like the default in Calcite [is in fact `EXCEPT DISTINCT`](https://github.com/apache/calcite/blob/master/core/src/main/codegen/templates/Parser.jj#L6011), and it's actually not mentioned on the calcite reference page. The line we were both referencing says that `MULTISET EXCEPT` defaults to all. It doesn't seem to specify `EXCEPT`'s default behavior anywhere.

I'm going to change this comment to `EXCEPT ALL` and `EXCEPT [DISTINCT]` and resolve this.